packages/mermaid/src/diagrams/pie/pieDb.ts

```diff
@@ -0,0 +1,106 @@
+import { log } from '../../logger.js';
+import { parseDirective as _parseDirective } from '../../directiveUtils.js';
+import { getConfig as commonGetConfig } from '../../config.js';
+import { sanitizeText } from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  setDiagramTitle,
+  getDiagramTitle,
+  getAccDescription,
+  setAccDescription,
+  clear as commonClear,
+} from '../../commonDb.js';
+import type { ParseDirectiveDefinition } from '../../diagram-api/types.js';
+import type { PieFields, PieDb, Sections, PieDiagramConfig } from './pieTypes.js';
+import type { RequiredDeep } from 'type-fest';
+
+export const DEFAULT_PIE_CONFIG: Required<PieDiagramConfig> = {
+  useMaxWidth: true,
+  useWidth: 1200,
+  textPosition: 0.75,
+} as const;
+
+export const DEFAULT_PIE_DB: RequiredDeep<PieFields> = {
```

Comment:
Yeah, I searched for how to make optional attributes defined, and I saw that `Required` is the way to go. Then I went to `type-fest` because I remembered once that they have a deep one, and they do, so I used it.

Is it a good approach? Because I need to make the global `reset`, `setConfig`, `getConfig`, and `pieRenderer` use DB ones, which affect a lot of parts.

Reply:
Yup, this seems like the right approach. `Required` is only the first level, so you do need this.
And we don't have to worry about bundle size as it's compile time only.

Reply:
Not sure how to make global `reset`, `setConfig`, `getConfig` updates the pie ones, I tried the following but it didn't work:

```ts
export const setConfig = (conf: MermaidConfig): MermaidConfig => {
  db.setConfig({ ...conf?.pie });
  conf.pie = db.getConfig();

  checkConfig(conf);
  assignWithDepth(currentConfig, conf);

  return getConfig();
};
```

If have time take look, and maybe resolve it.

Reply:
@sidharthv96 Should we use the diagram's config in API (`mermaidAPI.getConfig` and `mermaidAPI.setConfig` ...etc) in this PR, or leave for future PRs?